src/JSInterop/Microsoft.JSInterop/src/IJSRuntime.cs

```diff
@@ -36,5 +36,29 @@ public interface IJSRuntime
         /// <param name="args">JSON-serializable arguments.</param>
         /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
         ValueTask<TValue> InvokeAsync<TValue>(string identifier, CancellationToken cancellationToken, object[] args);
+
+        /// <summary>
+        /// Invokes the specified JavaScript function asynchronously.
+        /// <para>
+        /// <see cref="JSRuntime"/> will apply timeouts to this operation based on the value configured in <see cref="JSRuntime.DefaultAsyncTimeout"/>. To dispatch a call with a different timeout, or no timeout,
+        /// consider using <see cref="InvokeVoidAsync(string, CancellationToken, object[])" />.
+        /// </para>
+        /// </summary>
+        /// <param name="identifier">An identifier for the function to invoke. For example, the value <c>"someScope.someFunction"</c> will invoke the function <c>window.someScope.someFunction</c>.</param>
+        /// <param name="args">JSON-serializable arguments.</param>
+        /// <returns>A <see cref="ValueTask"/> that represents the asynchronous invocation operation.</returns>
+        ValueTask InvokeVoidAsync(string identifier, object[] args);
+
+        /// <summary>
+        /// Invokes the specified JavaScript function asynchronously.
+        /// </summary>
+        /// <param name="identifier">An identifier for the function to invoke. For example, the value <c>"someScope.someFunction"</c> will invoke the function <c>window.someScope.someFunction</c>.</param>
+        /// <param name="cancellationToken">
+        /// A cancellation token to signal the cancellation of the operation. Specifying this parameter will override any default cancellations such as due to timeouts
+        /// (<see cref="JSRuntime.DefaultAsyncTimeout"/>) from being applied.
+        /// </param>
+        /// <param name="args">JSON-serializable arguments.</param>
+        /// <returns>A <see cref="ValueTask"/> that represents the asynchronous invocation operation.</returns>
+        ValueTask InvokeVoidAsync(string identifier, CancellationToken cancellationToken, object[] args);
```

Comment:
Adding these methods to `IJSRuntime` would be a breaking change. If at all possible, we'd strive to avoid that.

Does it work to not add these, and instead only change the implementation of `JSRuntimeExtensions.InvokeVoidAsync`? If you changed the implementation of that extension method to call `jsRuntime.InvokeAsync<VoidReturn>` internally, then `JSRuntime.cs` could still be aware of that magic return type and use it as a signal not to deserialize the JSON, and just be hardcoded to return `default(VoidReturn)` which would be discarded by `JSRuntimeExtensions.InvokeVoidAsync`.